src/node.cc

```diff
@@ -3910,6 +3939,11 @@ static void StartNodeInstance(void* arg) {
           v8::platform::PumpMessageLoop(default_platform, isolate);
           EmitBeforeExit(env);
 
+          // We need to run an extra event loop turn to purge unrefed handles.
+          bool unrefedTimers = HasUnrefedTimerHandles(env);
+          if (unrefedTimers == true)
+            uv_run(env->event_loop(), UV_RUN_NOWAIT);
```

Comment:
Unfortunately this is too costly. This is an Object property lookup for every handle on every event loop. I'll think about how this could be done better.

Maybe @bnoordhuis has a better idea of how to take care of this.